pandas/core/arrays/numeric.py

```diff
@@ -190,6 +192,10 @@ def _coerce_to_data_and_mask(
         if values.dtype.kind in "iu":
             # fastpath
             mask = np.zeros(len(values), dtype=np.bool_)
+        elif values.dtype.kind == "f":
+            # np.isnan is faster than is_numeric_na() for floats
+            # github issue: #60066
+            mask = np.isnan(values)
```

Comment:
Could we ever get a masked float `values` here with `pd.NA`?

Reply:
There is a 

```
    if not copy:
        values = np.asarray(values)
    else:
        values = np.array(values, copy=copy)
```

above, so `values` is guaranteed to be a numpy array at this point